src/main/java/com/aloc/aloc/problem/service/ProblemSolvingService.java

```diff
@@ -82,4 +86,11 @@ public List<SolvedProblem> getSolvedProblemListByUser(Long userId) {
 	public int getSolvedCount(Long userId) {
 		return solvedProblemRepository.countByUserId(userId);
 	}
+
+	public void checkProblemIsSolvedAndAddSolvedProblem(User user, Problem problem)
```

Comment:
이 함수에는 문제를 풀었는지 체크하는 기능과, 풀었을 때 solvedProblem으로 저장하는 기능이 담겨있네요!
SOLID패턴 중 SRP를 고려하여 하나의 함수에는 하나의 기능만 담고있으면 추후 유지보수 시에 좋아요!

```suggestion
	public void updateUserProblemInfo(User user, Problem problem) {
		if(!checkProblemIsSolved) return;
		return AddSolvedProblem();
	}
```

Reply:
오 분리하는게 좋겠네요! 추후 userProblem 관련사항 머지 후 이 부분 까지 같이 해결하도록하겠습니다